### PR TITLE
[WIP] Update Blazor + Electron sample

### LINKED ISSUE #187: ComponentsElectron needs update to latest ASP.NET Core 3.0 preview
After upgrading .NET Core SDK from preview 3 to preview 4, I can no longer build the project.

`dotnet build` output

```
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 387.69 ms for C:\Users\user\Desktop\SampleApp\SampleApp.csproj.
C:\Program Files\dotnet\sdk\3.0.100-preview4-011223\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(151,5): message NETSDK1057: You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
Startup.cs(1,39): error CS0234: The type or namespace name 'Builder' does not exist in the namespace 'Microsoft.AspNetCore.Components' (are you missing an assembly reference?) [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
obj\Debug\netcoreapp3.0\RazorDeclaration\Shared\MainLayout.cshtml.g.cs(16,31): error CS0246: The type or namespace name 'LayoutComponentBase' could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
obj\Debug\netcoreapp3.0\RazorDeclaration\Pages\_ViewImports.cshtml.g.cs(16,69): error CS0246: The type or namespace name 'MainLayout' could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
obj\Debug\netcoreapp3.0\RazorDeclaration\Shared\MainLayout.cshtml.g.cs(19,33): error CS0115: 'MainLayout.BuildRenderTree(RenderTreeBuilder)': no suitable method found to override [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
Startup.cs(12,31): error CS0246: The type or namespace name 'IComponentsApplicationBuilder' could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]

Build FAILED.

Startup.cs(1,39): error CS0234: The type or namespace name 'Builder' does not exist in the namespace 'Microsoft.AspNetCore.Components' (are you missing an assembly reference?) [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
obj\Debug\netcoreapp3.0\RazorDeclaration\Shared\MainLayout.cshtml.g.cs(16,31): error CS0246: The type or namespace name 'LayoutComponentBase' could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
obj\Debug\netcoreapp3.0\RazorDeclaration\Pages\_ViewImports.cshtml.g.cs(16,69): error CS0246: The type or namespace name 'MainLayout' could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
obj\Debug\netcoreapp3.0\RazorDeclaration\Shared\MainLayout.cshtml.g.cs(19,33): error CS0115: 'MainLayout.BuildRenderTree(RenderTreeBuilder)': no suitable method found to override [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
Startup.cs(12,31): error CS0246: The type or namespace name 'IComponentsApplicationBuilder' could not be found (are you missing a using directive or an assembly reference?) [C:\Users\user\Desktop\SampleApp\SampleApp.csproj]
    0 Warning(s)
    5 Error(s)

Time Elapsed 00:00:08.11
PS C:\Users\user\Desktop\SampleApp>
```

### PR DESCRIPTION
**Pull request by danroth27**
@SteveSandersonMS How is the Microsoft.AspNetCore.Components.Electron package handled? Is it published to the dev feeds?

Fixes #187

### CODE REVIEW

File: src/ComponentsElectron/sample/SampleApp/SampleApp.csproj [hunk 1]
```diff
@@ -4,8 +4,8 @@
     <TargetFramework>netcoreapp3.0</TargetFramework>
     <OutputType>WinExe</OutputType>
     <LangVersion>7.3</LangVersion>
-    <ComponentsVersion>3.0.0-preview3-19153-02</ComponentsVersion>
-    <ComponentsElectronVersion>0.1.0-alpha.19165.3</ComponentsElectronVersion>
+    <ComponentsVersion>3.0.0-preview5-19227-01</ComponentsVersion>
+    <ComponentsElectronVersion>0.1.0-dev</ComponentsElectronVersion>
```
Comment (pranavkm) [inline src/ComponentsElectron/sample/SampleApp/SampleApp.csproj]: This is definitely incorrect. Maybe we need to get a build out and reference that here if the intent is for this to be a sample for users.

@aspnet/build  does this repo publishes packages? If so, what feed does it reference?

File: src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron/ElectronRenderer.cs [hunk 2]
```diff
@@ -94,8 +96,19 @@ public Task AddComponentAsync(Type componentType, string domElementSelector)
         /// <inheritdoc />
         protected override Task UpdateDisplayAsync(in RenderBatch batch)
         {
-            var bytes = MessagePackSerializer.Serialize(batch);
-            var base64 = Convert.ToBase64String(bytes);
+            string base64;
```
Comment (pranavkm) [inline src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron/ElectronRenderer.cs]: I thought I changed this?

File: src/ComponentsElectron/sample/SampleApp/SampleApp.csproj (hunk 1 above)
Comment (natemcmaster) [reply src/ComponentsElectron/sample/SampleApp/SampleApp.csproj]: Not anymore. It used to push to myget.org, but we have disabled all deployments to myget. Who needs to consume experimental packages? Is it sufficient to say the sample code is something you have to build yourself if you want to use it in your own app?

File: src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron/ElectronRenderer.cs (hunk 2 above)
Comment (danroth27) [reply src/ComponentsElectron/src/Microsoft.AspNetCore.Components.Electron/ElectronRenderer.cs]: Sorry, I need to rebase on master

File: src/ComponentsElectron/sample/SampleApp/SampleApp.csproj (hunk 1 above)
Comment (danroth27) [reply src/ComponentsElectron/sample/SampleApp/SampleApp.csproj]: > This is definitely incorrect

Right, that's why this is still a WIP

> Who needs to consume experimental packages? Is it sufficient to say the sample code is something you have to build yourself if you want to use it in your own app?

The goal is that you should able to clone this repo and then `dotnet run` this sample. This is for folks who want to experiment with Blazor+Electron, including internal teams.

**Comment by SteveSandersonMS** [comment]
> How is the Microsoft.AspNetCore.Components.Electron package handled? Is it published to the dev feeds?

I manually pushed it to the blazor-dev feed on MyGet (https://dotnet.myget.org/feed/blazor-dev/package/nuget/Microsoft.AspNetCore.Components.Electron). This is obviously not a permanent solution, or even something that would work if we were updating this regularly, but I think it's manageable for a "labs" package we only update every 1-2 months.

If there's a simple better way to automate this I'd be totally into that though!

**Comment by SteveSandersonMS** [comment]
Update: No, that was the old system before we moved to asplabs.

The current system is that the asplabs CI automatically pushed packages to the aspnetcore-dev feed on MyGet: https://dotnet.myget.org/feed/aspnetcore-dev/package/nuget/Microsoft.AspNetCore.Components.Electron

I guess that due to more recent changes with our processes, that is no longer happening like Nate comments above.

**Comment by natemcmaster** [comment]
@SteveSandersonMS do you have permissions to push manually to blazor-dev? That might be the simplest solution for now.

**Comment by natemcmaster** [comment]
:ping: @SteveSandersonMS do you have permissions to push manually to blazor-dev?

**Comment by SteveSandersonMS** [comment]
> do you have permissions to push manually to blazor-dev?

Yes, and I'm happy to do so! For flexibility, perhaps @eilon could also grant permission there to @danroth27?

**Comment by Eilon** [comment]
@SteveSandersonMS and @danroth27 already both have permission to push to blazor-dev.

**Comment by Eilon** [comment]
And now @natemcmaster does too, just in case.

**Comment by SteveSandersonMS** [comment]
Thanks!

**Comment by pranavkm** [comment]
Closing because this is super out of date.

**Pull request closed without merging**